core/src/smartcontracts/isi/world.rs

```diff
@@ -184,6 +226,102 @@ pub mod isi {
             })
         }
     }
+
+    impl Execute for Unregister<PermissionTokenDefinition> {
+        type Error = Error;
+
+        #[metrics("unregister_permission_token")]
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView,
+        ) -> Result<(), Self::Error> {
+            let definition_id = self.object_id;
+
+            remove_token_from_roles(wsv, &definition_id)?;
+            remove_token_from_accounts(wsv, &definition_id)?;
+
+            wsv.modify_world(|world| {
+                match world.permission_token_definitions.remove(&definition_id) {
+                    Some((_, definition)) => {
+                        Ok(PermissionTokenEvent::DefinitionDeleted(definition).into())
+                    }
+                    None => Err(FindError::PermissionTokenDefinition(definition_id).into()),
+                }
+            })?;
+
+            Ok(())
+        }
+    }
+
+    /// Removes all tokens with specified definition id from all registered roles
+    fn remove_token_from_roles(
+        wsv: &WorldStateView,
+        target_definition_id: &<PermissionTokenDefinition as Identifiable>::Id,
+    ) -> Result<(), Error> {
+        let mut roles_containing_token = Vec::new();
+
+        for role_entry in wsv.roles().iter() {
+            let (role_id, role) = role_entry.pair();
+            if role
+                .permissions()
+                .any(|token| token.definition_id() == target_definition_id)
+            {
+                roles_containing_token.push(role_id.clone())
+            }
+        }
+
+        for role_id in roles_containing_token {
+            wsv.modify_world(|world| match world.roles.get_mut(&role_id) {
+                Some(mut role) => {
+                    role.remove_permissions(target_definition_id);
+                    Ok(RoleEvent::Modified(role_id.clone()).into())
```

Comment:
I see, that it's used only here. Let's rename it to `PermissionRemoved` and add a permission token inside this event. May be it should look like this:

```rust
pub enum RoleEvent {
    // ...
    PermissionRemoved(PermissionRemoved)
}

pub struct PermissionRemoved {
    role_id: RoleId,
    permission_definition_id: PermissionDefinitionId,
}
```

I'm thinking about this, because now with [parametrised triggers](https://github.com/hyperledger/iroha/pull/2569) it may be useful for smart contract writers to know wat exactly have been changed